tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/Metrics.java

```diff
@@ -55,4 +55,52 @@
         tf.nn.inTopK(fPredictions, tf.math.argMax(labels, tf.constant(-1)), tf.constant(k)),
         predictions.type());
   }
+
+  /**
+   * Computes how often integer targets are in the top K predictions.
+   *
+   * <p>Standalone usage:
+   *
+   * <pre>
+   *     Operand&lt;TInt32&gt; labels = tf.constant(new int[]{2, 1});
+   *     Operand&lt;TFloat32&gt; predictions = tf.constant(new float[][]
+   *                            {{0.1f, 0.9f, 0.f8}, {0.05f, 0.95f, 0f}});
+   *     Operand&lt;TFloat32&gt; m = Metrics.topKCategoricalAccuracy(
+   *                                    labels, predictions, 3)
+   *     //m.shape().toString == "[2]"
+   * </pre>
+   *
+   * @param tf the TensorFlow Ops.
+   * @param labels the ground truth values.
+   * @param predictions The prediction values.
+   * @param k Number of top elements to look at for computing accuracy.
+   * @param <T> the data type for the predictions and results
+   * @param <U> the data type ofr the labels.
+   * @return the Operand for the Sparse top K categorical accuracy value.
+   */
+  @SuppressWarnings("unchecked")
+  public static <T extends TNumber, U extends TNumber> Operand<T> sparseTopKCategoricalAccuracy(
+      Ops tf, Operand<U> labels, Operand<T> predictions, int k) {
+    Operand<T> tLabels;
+    if (labels.type() != predictions.type())
+      tLabels = CastHelper.cast(tf, labels, predictions.type());
+    else tLabels = (Operand<T>) labels;
```

Comment:
The style guide says we should always have braces on if/else statements.

Reply:
Actually, I removed the if, as `cast` checks if the data types are different.